### ISSUE TITLE
[BUG] Network output is not same when changing the batch size during eval mode

### ISSUE THREAD

**Issue by zhangfuyang** [mit-han-lab/torchsparse#183]
### Is there an existing issue for this?

- [X] I have searched the existing issues

Hi, thank you for open source the code.

I have a problem that the network produces the different output when the batch size changes during the inference stage. A simple example to reproduce the issue.

Basically, I want to verify the similar rule as the regular network: `model([x1])` and `model([x1, x2])[0]` should always return the same output under `eval()` model, regardless the different batch size.

```
import torchsparse.nn as spnn
import torch.nn as nn
from torchsparse.utils.collate import sparse_collate
from torchsparse import SparseTensor
import torch

class Model(nn.Module):
    def __init__(self, feats_in=3):
        super().__init__()
        self.stem = nn.Sequential(
            spnn.Conv3d(feats_in, 32, kernel_size=3, stride=2),
            #spnn.BatchNorm(32),
            spnn.ReLU(True),
        )

    def forward(self, sp: SparseTensor):
        return self.stem(sp)

if __name__ == "__main__":

    for i in range(100):
        print(f'{i}:')
        feats_in = 3
        n = 10000
        model = Model(feats_in)
        model.cuda()
        model.eval()

        f1 = torch.randn(n, feats_in).cuda()
        c1 = torch.randint(0,100, (n,3), dtype=torch.int32).cuda()
        x1 = SparseTensor(coords=c1, feats=f1, stride=(1,1,1))

        f2 = torch.randn(n, feats_in).cuda()
        c2 = torch.randint(0,100, (n,3), dtype=torch.int32).cuda()
        x2 = SparseTensor(coords=c2, feats=f2, stride=(1,1,1))

        batch_input1 = sparse_collate([x1])
        batch_input2 = sparse_collate([x1, x2])

        # check
        o1 = model(batch_input1)
        o2 = model(batch_input2)

        print(o1.F.sum() == o2.F[o2.C[:,3]==0].sum())
```

### Expected Behavior

The expected behaivor should be always print **`True`**, meaning the model returns the exact same feature for the first batch. But it's not.

### Anything else?

An additional question is why the output point number is larger than the input when Conv3d's kernel_size is 3?

**Comment by skanti** [comment]
+1

(I've encountered the same bug)

**Comment by zhijian-liu** [comment]
Since TorchSparse has been upgraded to v2.1.0, could you please attempt to install the latest version? I will now close this issue, but please don't hesitate to reopen it if the problem persists.

**Comment by debOliveira** [comment]
I am also facing the same issue. Would you have any update on this @zhijian-liu @skanti @zhangfuyang?
